### DISCUSSION TITLE
BPF rendering error

### DISCUSSION THREAD

**Discussion by hacker0ni** [Security-Onion-Solutions/securityonion#5523] [Unsupported Versions] [answered]
I've read the documentation about BPFs and it looks like I can properly implement them using something like the following on the global.sls:
```
steno:
  bpf:
    - not port 443 &&
    - not host 192.168.1.24
```
The problem is, I've got a wide range of IP addresses that I specifically want to monitor while ignoring a lot of the IPs in the network, say 50 IP addresses that I want to monitor. I try to add them using something like this to the forward node's minion.sls under zeek, nids and steno:
```
zeek:
  bpf:
    - host 192.168.1.24 &&
    - host 192.168.5.36 &&
    - host 192.168.9.47 &&
    - host 192.168.9.48 &&
    .
    .
```
Suricata and Zeek complain : [ERROR   ] BPF Syntax Error - Discarding Specified BPF

When I use `not host 192.168.1.23` it works fine. But the amount of IPs to be ignored is too large. When I use something like `net 192.168.9` it drops all the packets. How can I make sure Zeek, Suricata and Stenographer only monitor specific IP addresses if I have multiple IP addresses?

**Comment by ben-sec** [comment]
I put my BPF statements in " " like
```
zeek:
  bpf:
    - "host 192.168.1.24 &&"
```
Maybe that does the trick ...

Cheers, Ben

**Reply by hacker0ni** [reply]
Thanks for the response Ben. I've tried encapsulating all of them in quotes, still no dice. Not sure if I'm doing something wrong. Let's say I want to monitor 192.168.0.1-3 and 192.168.0.7-8, but skip 192.168.0.4-6. What I do is I append the following under zeek and nids:
```
zeek:
  bpf:
    - "host 192.168.0.1 &&"
    - "host 192.168.0.2 &&"
    - "host 192.168.0.3 &&"
    - "host 192.168.0.7 &&"
    - "host 192.168.0.8"
```
When I restart zeek, suricata and steno they all complain about BPF error, discarding. When I use "net 192.168.0" it doesn't complain but then all the tools drop every packet haha. I can't wrap my head around this thing.

**Comment by dougburks** [comment, marked as answer]
Try changing `&&` to `||`. This changes the `and` to an `or`.

**Reply by hacker0ni** [reply]
Looks like that's it! Thanks Doug! The documentation said to use && at the end of every line which made me think of it as a rule appending mechanism instead of an `and` operand. For those of you who had problems with the same issue, my current config looks like this on the minion.sls (forward node):
```
zeek:
  bpf:
    - "host 192.168.0.1 ||"
    - "host 192.168.0.2 ||"
    - "host 192.168.0.7 ||"
    - "host 192.168.0.8 ||"
    .
    .
    .
```
I'd highly appreciate it if you could mention this in the official documentation as well. Thanks again!

**Reply by dougburks** [reply]
Updated documentation:
https://docs.securityonion.net/en/2.3/bpf.html#multiple-conditions

**Comment by hacker0ni** [comment]
It looks like I can at least limit some of the traffic using `not net 192.168.1`. However there is an issue with filtering by including hosts, it's either my deployment or it's a general bug. If I use `net 192.168.0` idstools will not complain but drop every traffic. If I use `host 192.168.0.1` idstools will not even parse the BPF. My current configuration that works looks something like this, but it's not optimal:
```
steno:
  bpf:
    - not port 443 &&
    - not net 192.168.0 &&
    - not net 192.168.3
 zeek:
  bpf:
    - not net 192.168.0 &&
    - not net 192.168.3
```
etc. etc.
@dougburks Could you guys recreate this issue and see if this is a bug or something wrong with my deployment?
